src/lib.rs

```diff
@@ -139,21 +197,39 @@ fn run_fixing_loop(
         } else {
             violations.clear();
         }
+        let mut instantiated_per_file_rules: HashMap<RuleName, Arc<dyn RuleInstancePerFile>> =
```

Comment:
So then inside eg the fixing loop we can have a simpler version of the "per-file instantiated rules" data structure (because we're in the context of a single known file path and running serially wrt ourselves)